src/realm/error_codes.yml

```diff
@@ -0,0 +1,48 @@
+error_categories:
+  - generic_error
+  - logic_error
+  - runtime_error
+  - invalid_argument
+
+error_codes:
```

Comment:
Shouldn't all of these errors be assigned a category?

Reply:
This file is not up-to-date. I am not sure how this will look eventually.